flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -224,7 +224,7 @@ public DynamicTableSink createDynamicTableSink(Context context) {
                 keyProjection,
                 valueProjection,
                 keyPrefix,
-                tableOptions.get(TOPIC).get(0),
+                tableOptions.getOptional(TOPIC).isPresent() ? tableOptions.get(TOPIC).get(0) : null,
```

Comment:
@twalthr , this list couldn't be empty because this case is checked in `KafkaOptions#validateSinkTopic` method.

Reply:
Here at this place, you could use `unspecified!` instead of null. Then you don't need to change anything in the `KafkaProducer`. If you use a special character like `!`, this will fail hard if the user row does not specify the column.

Reply:
@AHeise , I agree with this approach but I'm confused about why not this uses null? I don't understand the difference between `unspecified!` and null.